apps/libs/rootserver/src/index.ts

```diff
@@ -0,0 +1,35 @@
+import { ConfigService } from '@nestjs/config'
+import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify'
+
+export abstract class RootServer {
+  app?: NestFastifyApplication
+  adapter = new FastifyAdapter({ logger: true })
+
+  abstract create (): Promise<NestFastifyApplication>
+
+  async configure (app: NestFastifyApplication, config: ConfigService): Promise<void> {
+    app.enableCors({
+      origin: '*',
+      methods: ['GET', 'PUT', 'POST', 'DELETE'],
+      allowedHeaders: ['Content-Type'],
+      maxAge: 60 * 24 * 7
+    })
+  }
+
+  async init (app: NestFastifyApplication, config: ConfigService): Promise<void> {
+    const port = config.get<number>('PORT', 3000)
+    await app.listen(port, '0.0.0.0')
+  }
+
+  async start (): Promise<void> {
+    this.app = await this.create()
```

Comment:
Since we are assigning a new application instance to this variable. Should we be running `stop` before we create a new instance, in case another instance is already running?
```suggestion
  async start (): Promise<void> {
    await this.stop()
    this.app = await this.create()
```